### PR TITLE
Make LoggerUsageTask cacheable

### PR DESCRIPTION
**Pull request by mark-vieira**
Make the `LoggerUsageTask` cacheable. Also remove some unnecessary complexity in some of the implementation logic and other misc cleanup.

### CODE REVIEW

File: buildSrc/src/main/java/org/elasticsearch/gradle/precommit/PrecommitTask.java [hunk 1]
```diff
@@ -36,7 +36,6 @@ public File getSuccessMarker() {
 
     @TaskAction
     public void writeMarker() throws IOException {
-        getSuccessMarker().getParentFile().mkdirs();
```
Comment (rjernst) [inline buildSrc/src/main/java/org/elasticsearch/gradle/precommit/PrecommitTask.java]: Why is this removed? Don't we need to ensure the parent dir exists?

**Review by rjernst — APPROVED** [review]
Looks fine, one question

File: buildSrc/src/main/java/org/elasticsearch/gradle/precommit/PrecommitTask.java (hunk 1 above)
Comment (mark-vieira) [reply buildSrc/src/main/java/org/elasticsearch/gradle/precommit/PrecommitTask.java]: Good question. This is actually unnecessary because annotating `getSuccessMarker()` with `@OutputFile` already ensures that file and any parent directories will be created for us.
